### PR TITLE
boards: arm: add ruuvi_ruuvitag

### PR DESCRIPTION
**Pull request by ghost**
Adds ARM based Ruuvtag baord. This board is based on the nRF52832.

Signed-off-by: the BASTION theBASTI0N <thebasti0ncode@gmail.com>

@carlescufi Thanks for the help earlier. I think this should have fixed the issue.

I have mentioned the flashing option via SWDIO and SWDCLK pin but did not go into detail.

### CODE REVIEW

**Comment by theBASTI0N** [comment]
Scancode / Scan code for licenses (pull_request) is failing on Kconfig.defconfig file. 

Not sure why as it has the license header.

**Comment by carlescufi** [comment]
@theBASTI0N you need to squash the 3 commits together and then force-push. You also need to follow the commit formatting guidelines I linked you to.

**Comment by theBASTI0N** [comment]
@carlescufi Not sure why the last two checks are failing.

File: boards/arm/ruuvi_ruuvitag/Kconfig
Comment (jukkar) [inline boards/arm/ruuvi_ruuvitag/Kconfig:1]: Small typo in commit message body, s/baord/board/

**Review by carlescufi — APPROVED** [review]
Thank you @theBASTI0N.

**Comment by carlescufi** [comment]
> @carlescufi Not sure why the last two checks are failing.

They are all passing now, so this is ready once you get an additional approval. Note  that this will have to wait until 2.3 is released since the feature merge window is currently closed.

**Comment by theBASTI0N** [comment]
@carlescufi Thanks for the help.

File: boards/arm/ruuvi_ruuvitag/Kconfig
Comment (theBASTI0N) [reply boards/arm/ruuvi_ruuvitag/Kconfig:1]: Thanks, I'll fix it shortly

Comment (theBASTI0N) [reply boards/arm/ruuvi_ruuvitag/Kconfig]: @jukkar typo has been corrected.

**Comment by carlescufi** [comment]
@ioannisg @anangl can you review please?

File: doc/conf.py
Comment (ioannisg) [inline doc/conf.py:407]: This is not needed, since this is is a new board. Remove (unless I am missing something)

**Review by ioannisg — CHANGES_REQUESTED** [review]

File: doc/conf.py
Comment (carlescufi) [reply doc/conf.py:407]: That's right, you can remove that line.

File: boards/arm/ruuvi_ruuvitag/ruuvi_ruuvitag.dts
Comment (pabigot) [inline boards/arm/ruuvi_ruuvitag/ruuvi_ruuvitag.dts:12]: This should be `ruuvi,ruuvitag` (manufacturer,model) since the product is `RuuviTag` (capitalized that way; please check the PR for `Ruuvitag` and fix it).

**Review by pabigot — CHANGES_REQUESTED** [review]
I've been meaning to get a couple of these now that the devkit is back in stock; thanks for submitting the board support.

One change request for the devicetree compatible.

File: doc/conf.py
Comment (theBASTI0N) [reply doc/conf.py:407]: Ok thanks, wasn't sureb about that.

File: boards/arm/ruuvi_ruuvitag/ruuvi_ruuvitag.dts
Comment (theBASTI0N) [reply boards/arm/ruuvi_ruuvitag/ruuvi_ruuvitag.dts]: Thanks that has been change now.

File: boards/arm/ruuvi_ruuvitag/ruuvi_ruuvitag.dts
Comment (theBASTI0N) [reply boards/arm/ruuvi_ruuvitag/ruuvi_ruuvitag.dts]: Thanks for picking that up. I have corrected it now.

File: boards/arm/ruuvi_ruuvitag/Kconfig.board
Comment (theBASTI0N) [reply boards/arm/ruuvi_ruuvitag/Kconfig.board]: Thanks for pointing that out. I remove that.

File: doc/conf.py
Comment (theBASTI0N) [reply doc/conf.py]: I removed that file.

**Comment by theBASTI0N** [comment]
Would there be a reason Shappable is now failing at:
image
Also I am not sure about buildkite either that seems new.

**Review by ioannisg — APPROVED** [review]
Looks OK now, thanks

**Review by anangl — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Comment by pabigot** [comment]
@theBASTI0N did you test the UART configuration on this board?  I've got some rev B7.1 boards, and am having no luck, even though I've confirmed that the selected RX/TX pins are correct by configuring them as LEDs.  Very strange.

**Comment by pabigot** [comment]
Looks like missing default configs: the following are normally in board_defconfig, but of course aren't here because this board doesn't normally have a console.  Leaving any one of them off silently prevents console output from appearing.

```
CONFIG_SERIAL=y
CONFIG_CONSOLE=y
CONFIG_UART_CONSOLE=y
```

**Pull request merged**
